### PR TITLE
owasp action

### PR DESCRIPTION
**Pull request by acouch**
## Ticket

Resolves [FFS-1083](https://jiraent.cms.gov/browse/FFS-1083)

## Changes

Make the OWASP running actually run. Started: https://github.com/DSACMS/iv-cbv-payroll/pull/116

This adds a couple of ignore: 

```
10110	IGNORE	Dangerous JS Functions
# There are necessary and properly scoped in content_security_policy.rb
10055	IGNORE	CSP: Notices
# Perms Policy not supported in rails https://github.com/rails/rails/issues/48878
# Also not fully adopted: https://caniuse.com/permissions-policy
10063	IGNORE	Permissions Policy Header Not Set
```

Unfortunately, [Zap does not support ignoring risk levels like LOW](https://groups.google.com/g/zaproxy-users/c/Vdan6QRHfRM), so ignoring a full check is the only way to filter.

### CODE REVIEW

File: .github/workflows/owasp-daily-scan.yml [hunk 1]
```diff
@@ -1,10 +1,12 @@
 name: OWASP ZAP daily scan
 
 on:
-  schedule:
-    # cron format: 'minute hour dayofmonth month dayofweek'
-    # this will run at noon UTC every day (7am EST / 8am EDT)
-    - cron: '0 12 * * *'
+  pull_request:
+    branches: [main]
```
Comment (tdooner) [inline .github/workflows/owasp-daily-scan.yml:5]: nit: File is misnamed if this is no longer a daily scan

File: app/app/controllers/pages_controller.rb [hunk 2]
```diff
@@ -1,4 +1,7 @@
 class PagesController < ApplicationController
   def home
+    unless params["format"].nil?
+      head 401, content_type: "text/html"
+    end
```
Comment (tdooner) [inline app/app/controllers/pages_controller.rb:5]: I wonder if this is still necessary given your change to the `routes.rb` you've discussed in Slack

**Review by tdooner — APPROVED** [review]

File: .github/workflows/owasp-daily-scan.yml (hunk 1 above)
Comment (acouch) [reply .github/workflows/owasp-daily-scan.yml:5]: Good catch, I renamed to `owasp-scan.yml` so the name is separate from the trigger.

File: app/app/controllers/pages_controller.rb (hunk 2 above)
Comment (acouch) [reply app/app/controllers/pages_controller.rb:5]: This was introduced in https://github.com/DSACMS/iv-cbv-payroll/pull/116 . 

@joeyg was there a reason this was added? I'm reading it as returning unauthorized if a request to the home page doesn't have a request format.

Comment (joeyg) [reply app/app/controllers/pages_controller.rb:5]: @acouch One of the checks is for files like ._hg and the like.

Comment (acouch) [reply app/app/controllers/pages_controller.rb:5]: @joeyg Thanks, I can confirm that https://github.com/DSACMS/iv-cbv-payroll/pull/124 fixes that. I'm going to remove from here and rebase once #124 is pulled in.

**Pull request merged**